MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift

```diff
@@ -126,31 +104,178 @@ private extension FarmerDetailsView {
                 }
                 Spacer()
                 directionButton
-                    .frame(width: 80, height: 35)
-                    .background(.ultraThinMaterial)
-                    .clipShape(.capsule)
-                    .padding()
-                    .shadow(radius: 2)
             }
             Divider()
-            Text("Téléphone")
-                .foregroundStyle(.secondary)
-                .font(.subheadline)
-            Text(viewModel.phoneNumber ?? "")
-                .font(.callout)
+            HStack {
+                VStack(alignment: .leading) {
+                    Text("Téléphone")
+                        .foregroundStyle(.secondary)
+                        .font(.subheadline)
+                    Text(viewModel.phoneNumber ?? "Aucun numéro disponible")
+                        .font(.callout)
+                }
+                Spacer()
+                phoneButton
+            }
             Divider()
-            Text("Site Web")
-                .foregroundStyle(.secondary)
-                .font(.subheadline)
+            HStack {
+                VStack(alignment: .leading) {
+                    Text("Email")
+                        .foregroundStyle(.secondary)
+                        .font(.subheadline)
+                    Text(viewModel.email ?? "Aucune adresse email")
+                        .font(.callout)
+                }
+                Spacer()
+                emailButton
+            }
+            Divider()
+            VStack(alignment: .leading) {
+                Text("Site Web")
+                    .foregroundStyle(.secondary)
+                    .font(.subheadline)
+                if !viewModel.websites.isEmpty {
+                    ForEach(viewModel.websites, id: \.id) { website in
+                        Text(.init(viewModel.formatWebsite(website)))
+                    }
+                } else {
+                    Text("Aucun site web")
+                        .font(.callout)
+                }
+            }
         }
         .padding()
         .background(.ultraThinMaterial)
         .clipShape(.rect(cornerRadius: 20))
-        .shadow(radius: 4)
+        .shadow(radius: 8)
+    }
+
+    private var directionButton: some View {
+        Button {
+            viewModel.onDirectionButtonTapped()
+        } label: {
+            Image(systemName: viewModel.directionButtonImageSystemName)
+        }
+        .frame(width: 80, height: 40)
+        .overlay(RoundedRectangle(cornerRadius: 32)
+            .stroke(.blue, lineWidth: 1))
+        .padding()
+    }
+
+    private var phoneButton: some View {
+        VStack {
+            if viewModel.phoneNumber != nil {
+                Link(destination: viewModel.phoneCallURL!) {
```

Comment:
You shouldn't use implicit unwrap. Do you really want to crash the app if the URL is wrong?

Reply:
Never!